src/libra_py/tsh.py

```diff
@@ -955,6 +955,8 @@ def hopping(Coeff, Hvib, istate, sh_method, do_collapse, ksi, ksi2, dt, T, boltz
     old_st = istate
     new_st = hop(istate, g, ksi)
 
+    Coeff1 = Coeff
```

Comment:
this is a shallow copy (by reference) - if you change Coeff1, it changes Coeff. Better copy using the matrix copy constructor